src/EditorFeatures/Core/Editor/EditorLayerExtensionManager.cs

```diff
@@ -5,126 +5,123 @@
-    internal class EditorLayerExtensionManager(
-        IGlobalOptionService optionService,
-        [ImportMany] IEnumerable<IExtensionErrorHandler> errorHandlers) : IWorkspaceServiceFactory
+    private readonly List<IExtensionErrorHandler> _errorHandlers = errorHandlers.ToList();
+
+    public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
     {
-        private readonly List<IExtensionErrorHandler> _errorHandlers = errorHandlers.ToList();
-        private readonly IGlobalOptionService _optionService = optionService;
+        var errorReportingService = workspaceServices.GetRequiredService<IErrorReportingService>();
+        var errorLoggerService = workspaceServices.GetRequiredService<IErrorLoggerService>();
+        return new ExtensionManager(errorReportingService, errorLoggerService, _errorHandlers);
+    }
 
-        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
-        {
-            var errorReportingService = workspaceServices.GetRequiredService<IErrorReportingService>();
-            var errorLoggerService = workspaceServices.GetRequiredService<IErrorLoggerService>();
-            return new ExtensionManager(_optionService, errorReportingService, errorLoggerService, _errorHandlers);
-        }
+    internal sealed class ExtensionManager(
+        IErrorReportingService errorReportingService,
+        IErrorLoggerService errorLoggerService,
+        List<IExtensionErrorHandler> errorHandlers) : AbstractExtensionManager
+    {
+        private readonly List<IExtensionErrorHandler> _errorHandlers = errorHandlers;
+        private readonly IErrorReportingService _errorReportingService = errorReportingService;
+        private readonly IErrorLoggerService _errorLoggerService = errorLoggerService;
 
-        internal class ExtensionManager(
-            IGlobalOptionService globalOptions,
-            IErrorReportingService errorReportingService,
-            IErrorLoggerService errorLoggerService,
-            List<IExtensionErrorHandler> errorHandlers) : AbstractExtensionManager
+        protected override void HandleNonCancellationException(object provider, Exception exception)
         {
-            private readonly List<IExtensionErrorHandler> _errorHandlers = errorHandlers;
-            private readonly IGlobalOptionService _globalOptions = globalOptions;
-            private readonly IErrorReportingService _errorReportingService = errorReportingService;
-            private readonly IErrorLoggerService _errorLoggerService = errorLoggerService;
+            Debug.Assert(exception is not OperationCanceledException);
 
-            public override void HandleException(object provider, Exception exception)
+            if (provider is CodeFixProvider
+                or CodeRefactoringProvider
+                or CodeRefactorings.FixAllProvider
+                or CodeFixes.FixAllProvider
+                or CompletionProvider)
```

Comment:
added CompletionProvider here.  That way this participates in teh yellow-bar, telling the user about the completion provider that thrw the exception and if htey want to enable it.